sqlalchemy-stubs/sql/elements.pyi

```diff
@@ -20,24 +31,24 @@ class ClauseElement(Visitable):
     description: Any = ...
     def unique_params(self, *optionaldict, **kwargs): ...
     def params(self, *optionaldict, **kwargs): ...
-    def compare(self, other, **kw): ...
-    def get_children(self, **kwargs): ...
-    def self_group(self, against: Optional[Any] = ...): ...
-    def compile(self, bind: Optional[Any] = ..., dialect: Optional[Any] = ..., **kw): ...
+    def compare(self, other: Any, **kw: Any) -> bool: ...
+    def get_children(self, **kwargs: Any) -> Any: ...
+    def self_group(self, against: Optional[Any] = ...) -> Any: ...
+    def compile(self, bind: Optional[Any] = ..., dialect: Optional[Any] = ..., **kw) -> Any: ...
     def __and__(self, other): ...
     def __or__(self, other): ...
     def __invert__(self): ...
     def __bool__(self): ...
     __nonzero__: Any = ...
 
-class ColumnElement(operators.ColumnOperators, ClauseElement):
+class ColumnElement(operators.ColumnOperators, ClauseElement, Generic[_T]):
     __visit_name__: str = ...
     primary_key: Any = ...
-    foreign_keys: Any = ...
+    foreign_keys: List[Any] = ...
     key: Any = ...
     def self_group(self, against: Optional[Any] = ...): ...
     @property
-    def type(self): ...
+    def type(self) -> _T: ...
```

Comment:
Can `_T` take any value here like `int` or `str`? If not, then maybe we should either add `bound=TypeEngine[Any]` to the type variable or just switch to using `TypeEngine[_T]` here?

Reply:
You're right. I'll update to a bound generic. Good catch!